contracts/cw721-stake/src/contract.rs

```diff
@@ -0,0 +1,1536 @@
+        .owner
+        .as_ref()
+        .map(|owner| match owner {
+            Owner::Addr(address) => deps.api.addr_validate(address),
+            Owner::Instantiator {} => Ok(info.sender),
+        })
+        .transpose()?;
+    let manager = msg
+        .manager
+        .as_ref()
+        .map(|h| deps.api.addr_validate(h))
+        .transpose()?;
+
+    let config = Config {
+        owner: owner.clone(),
+        manager,
+        nft_address: deps.api.addr_validate(&msg.nft_address)?,
+        unstaking_duration: msg.unstaking_duration,
+    };
+    CONFIG.save(deps.storage, &config)?;
+    TOTAL_STAKED_NFTS.save(deps.storage, &Uint128::zero(), env.block.height)?;
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    Ok(Response::default()
+        .add_attribute("method", "instantiate")
+        .add_attribute("nft_contract", msg.nft_address)
+        .add_attribute(
+            "owner",
+            owner
+                .map(|a| a.into_string())
+                .unwrap_or_else(|| "None".to_string()),
+        )
+        .add_attribute("manager", msg.manager.unwrap_or_else(|| "None".to_string())))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response<Empty>, ContractError> {
+    match msg {
+        ExecuteMsg::ReceiveNft(msg) => execute_stake(deps, env, info, msg),
+        ExecuteMsg::Unstake { token_ids } => execute_unstake(deps, env, info, token_ids),
+        ExecuteMsg::ClaimNfts {} => execute_claim_nfts(deps, env, info),
+        ExecuteMsg::UpdateConfig {
+            owner,
+            manager,
+            duration,
+        } => execute_update_config(info, deps, owner, manager, duration),
+        ExecuteMsg::AddHook { addr } => execute_add_hook(deps, env, info, addr),
+        ExecuteMsg::RemoveHook { addr } => execute_remove_hook(deps, env, info, addr),
+    }
+}
+
+pub fn execute_stake(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    wrapper: Cw721ReceiveMsg,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    if info.sender != config.nft_address {
+        return Err(ContractError::InvalidToken {
+            received: info.sender,
+            expected: config.nft_address,
+        });
+    }
+
+    let sender = deps.api.addr_validate(&wrapper.sender)?;
+    STAKED_NFTS_PER_OWNER.update(
+        deps.storage,
+        &sender,
+        env.block.height,
+        |nft_collection| -> StdResult<HashSet<String>> {
+            let mut updated_nft_collection = nft_collection.unwrap_or_default();
+            updated_nft_collection.insert(wrapper.token_id.clone());
+            Ok(updated_nft_collection)
+        },
+    )?;
+
+    TOTAL_STAKED_NFTS.update(
+        deps.storage,
+        env.block.height,
+        |total_staked| -> StdResult<_> {
+            total_staked
+                .unwrap()
```

Comment:
nit: with my above change this could simply be `unwrap_or_default` and then we can remove the setting to 0